comments/models.py

```diff
@@ -26,30 +29,57 @@ class CommentableMixin(models.Model):
 
     Use comments_panels to modify commenting_status from the admin.
     """
-    commenting_status = models.CharField(max_length=15, choices=CommentStatus.Choices)
+    commenting_status = models.CharField(max_length=15, choices=CommentStatus.Choices, default=CommentStatus.INHERITED)
     commenting_starts_at = models.DateTimeField(null=True, blank=True)
     commenting_ends_at = models.DateTimeField(null=True, blank=True)
 
-
     comments_panels = [
         FieldPanel('commenting_status', heading='Status'),
         FieldPanel('commenting_starts_at', heading='Commenting Opens At'),
         FieldPanel('commenting_ends_at', heading='Commenting Ends At')
 
     ]
 
+    def _get_valid_ancestor_for_commenting(self):
+        from home.models import Section
+
+        ancestors = self.get_ancestors().type(Section).specific().order_by('-path')
+        for ancestor in ancestors:
+            if ancestor.commenting_status in [
+                CommentStatus.OPEN, CommentStatus.CLOSED, CommentStatus.DISABLED, CommentStatus.TIMESTAMPED]:
+                return ancestor
+
     def should_show_comments_list(self):
-        return self.commenting_status in [CommentStatus.OPEN, CommentStatus.CLOSED, CommentStatus.TIMESTAMPED]
+        from home.models import Section
+
+        commenting_status = self.commenting_status
+        if commenting_status == CommentStatus.INHERITED:
+            ancestor = self._get_valid_ancestor_for_commenting()
+            commenting_status = ancestor.commenting_status
```

Comment:
Check for None, e.g. if the whole ancesteor tree has INHERITED, the page crashes.
In that case, use 'closed' as a default. (Or check what V1 uses, might be not so easy to find out though.)

Reply:
v1 was using disabled for it